request-server.go

```diff
@@ -173,8 +174,18 @@ func (rs *RequestServer) packetWorker(
 			rpkt = cleanPacketPath(pkt)
 		case *sshFxpOpendirPacket:
 			request := requestFromPacket(ctx, pkt)
-			handle := rs.nextRequest(request)
-			rpkt = sshFxpHandlePacket{pkt.id(), handle}
+			rpkt = request.call(rs.Handlers, pkt)
+			request.close()
```

Comment:
Just more curious, is there a reason to close the request here, and then make a new one to use for the handle? Seems like the original could be used instead of explicitly closing it to just make it again.

Reply:
It is the quickest/cleanest way to reset the Request back to 0 values. The Stat call for the Opendir and the (usually) following Readdir would both use the same Request.state and if it isn't reset they conflict.

Really Stat and Readdir shouldn't conflict this way as it is legal for a client to do this. I'll need to test if it is an actual bug though, as it might not be in practice (when a client is calling vs internal use). I'll make a note to come up with a test for this. I don't have time ATM.

Thanks.